packages/web3-eth-contract/src/utils.ts

```diff
@@ -0,0 +1,117 @@
+import { TransactionCall, TransactionWithSender } from 'web3-common';
+import { AbiFunctionFragment } from 'web3-eth-abi';
+import { mergeDeep } from 'web3-utils';
+import { encodeMethodABI } from './encoding';
+import { NonPayableCallOptions, PayableCallOptions, ContractOptions } from './types';
+
+export const getSendTxParams = ({
+	abi,
+	params,
+	options,
+	contractOptions,
+}: {
+	abi: AbiFunctionFragment;
+	params: unknown[];
+	options?: PayableCallOptions | NonPayableCallOptions;
+	contractOptions: ContractOptions;
+}): TransactionWithSender & { data: string } => {
+	if (!options?.to && !contractOptions.address) {
+		throw new Error('Contract address not specified');
+	}
+
+	if (!options?.from && !contractOptions.from) {
+		throw new Error('Contract "from" address not specified');
+	}
+
+	let txParams = mergeDeep(
+		{
+			to: contractOptions.address,
+			gas: contractOptions.gas,
+			gasPrice: contractOptions.gasPrice,
```

Comment:
Should also support EIP1559 type params, or create TODO new issue

Reply:
https://github.com/ChainSafe/web3.js/issues/4685